Scripts/GUI/GUIObjects/GUIInventorySlot.cs

```diff
@@ -47,11 +47,11 @@ public GUIInventorySlot(GUIInventorySlot toExchangeOnPress, Item.ItemType t, int
 
     public override void OnLeftPress()
     {
-        ItemStack floatingStack = exchangeSlot.GetCurItemStack();
-        ItemStack slotStack = this.GetCurItemStack();
+        ItemStack exchangeStack = exchangeSlot.GetCurItemStack();
+        ItemStack slotStack = GetCurItemStack();
 
-        if (floatingStack != null &&
-            !Item.CompatibleWith(floatingStack.Item.IType, this.GetItemType()))
+        if (exchangeStack != null &&
+            !Item.CompatibleWith(exchangeStack.Item.IType, this.GetItemType()))
```

Comment:
This could just be `!Item.CompatibleWith(exchangeStack?.Item.IType, this.GetItemType())` I think.

Reply:
Maybe not actually, that might cause the first term to return null, but the ItemType enum can't be null, hmmm

Reply:
It doesn't work like that, unfortunately 😞 

C# can't propagate null values (or as I like to call them the Maybe monad) through function calls and that's what would be necessary here, I think.